api/src/controllers/publications.js

```diff
@@ -150,6 +154,158 @@ async function readAllPublicationsByTeam(req, res) {
     res.status(200).json(foundPublication);
 }
 
+/**
+ * Given a google scholar user id, this function calls an AWS Lambda Python function that scrapes the publications off the user's profile and returns them.
+ * Some extra processing is done by this function to make the publications returned fit the publication model of the DB.
+ * Endpoint: /publications/import/:id
+ * @param req request object - google scholar user id given in the url
+ * @param res response object - array of publication objects
+ */
+async function getGoogleScholarPublications(req, res) {
+    let author = req.params.id;
+    console.log(author);
+
+    const client = lambda;
+    const params = {
+        FunctionName: 'import-pub-opt',
+        InvocationType: 'RequestResponse',
+        Payload: "{\"author_id\": \""+ author+"\"}"
+    };
+    const command = new InvokeCommand(params);
+    const response = await client.send(command);
+
+    const asciiDecoder = new TextDecoder('utf-8');
```

Comment:
Got this error when running it:
```
/home/akshay/.nvm/versions/node/v16.1.0/bin/node /home/akshay/.nvm/versions/node/v16.1.0/lib/node_modules/npm/bin/npm-cli.js run start --scripts-prepend-node-path=auto

> api@1.0.0 start
> node src/server.js

info: Server running on port: 5000 {"timestamp":"2021-05-22 18:26:53"}
info: Successfully connected to Atlas. {"timestamp":"2021-05-22 18:26:53"}
4BP_HvoAAAAJ
(node:21257) UnhandledPromiseRejectionWarning: ReferenceError: TextDecoder is not defined
    at getGoogleScholarPublications (/home/akshay/Desktop/Uni/FIT3170/Researchify/api/src/controllers/publications.js:177:26)
    at process._tickCallback (internal/process/next_tick.js:68:7)
(node:21257) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 10)
(node:21257) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```
it seems that `TextDecoder` is global, but strange that I got an error.

Reply:
Also tried with a different version of node:
```
akshay@death3ater:~/Desktop/Uni/FIT3170/Researchify/api$ npm start

> api@1.0.0 start
> node src/server.js

info: Server running on port: 5000 {"timestamp":"2021-05-22 18:30:08"}
info: Successfully connected to Atlas. {"timestamp":"2021-05-22 18:30:08"}
4BP_HvoAAAAJ
undefined:1
[object Object]
 ^

SyntaxError: Unexpected token o in JSON at position 1
    at JSON.parse (<anonymous>)
    at getGoogleScholarPublications (/home/akshay/Desktop/Uni/FIT3170/Researchify/api/src/controllers/publications.js:180:31)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
```
seems to be a different error this time.

Reply:
For reference here is the author id i'm using: `4BP_HvoAAAAJ`

Reply:
Will put these in the backlog to look at later